services/src/main/java/org/exoplatform/leadcapture/rest/LeadsManagementRest.java

```diff
@@ -0,0 +1,281 @@
+
+  private LeadCaptureSettingsService leadCaptureSettingsService;
+
+  public LeadsManagementRest(LeadsManagementService leadsManagementService,
+                             LeadCaptureSettingsService leadCaptureSettingsService) {
+    this.leadsManagementService = leadsManagementService;
+    this.leadCaptureSettingsService = leadCaptureSettingsService;
+  }
+
+  @OPTIONS
+  @Path("leads")
+  public Response allowCORS(@Context UriInfo uriInfo) throws Exception {
+    LeadCaptureSettings settings = leadCaptureSettingsService.getSettings();
+    Response.ResponseBuilder response = Response.ok();
+    response.header("Access-Control-Allow-Origin", settings.getAllowedCaptureSourceDomain());
+    response.header("Access-Control-Allow-Headers", "*");
+    response.header("Access-Control-Request-Method", "POST");
+    return response.build();
+  }
+
+  @GET
+  @Path("leads")
+  @Produces(MediaType.APPLICATION_JSON)
+  @RolesAllowed("ux-team")
+  public Response getLeads(@Context UriInfo uriInfo) throws Exception {
+    Identity sourceIdentity = Util.getAuthenticatedUserIdentity(portalContainerName);
+    if (sourceIdentity == null) {
+      return Response.status(Response.Status.UNAUTHORIZED).build();
+    }
+    try {
+      return Response.ok(leadsManagementService.getLeads()).build();
+    } catch (Exception e) {
+      LOG.error("An error occured when trying to get leads list", e);
+      return Response.status(Response.Status.INTERNAL_SERVER_ERROR).entity(e.getMessage()).build();
+    }
+  }
+
+  @POST
+  @Consumes(MediaType.APPLICATION_JSON)
+  @Path("leads")
+  public Response add(@Context UriInfo uriInfo, @HeaderParam("token") String headerToken, FormInfo lead) throws Exception {
+    LeadCaptureSettings settings = leadCaptureSettingsService.getSettings();
+/*    String captureToken  = System.getProperty(LEAD_CAPTURE_TOKEN);
+    if (headerToken == null) {
+      LOG.warn("Security Token for Lead capture not defined");
+      return Response.status(Response.Status.FORBIDDEN).entity("Access forbidden to the add lead rest service, Security Token not defined").build();
+    }
+    if (captureToken == null || !captureToken.equals(headerToken)) {
+      LOG.warn("Access forbidden to the add lead rest service, wrong token: {}", headerToken);
+      return Response.status(Response.Status.FORBIDDEN).entity("Access forbidden to the add lead rest service, wrong token").build();
+    }*/
+    if (lead == null) {
+      return Response.status(Response.Status.BAD_REQUEST).build();
+    }
+    if (lead.getLead() == null || StringUtils.isEmpty(lead.getLead().getMail())) {
+      return Response.status(Response.Status.BAD_REQUEST).entity("Lead mail needed").build();
+    }
+
+    if (!leadCaptureSettingsService.getSettings().isCaptureEnabled()) {
+      LOG.warn("Lead capture not enabled, New lead {} not captured ", lead.getLead().getId());
+      return Response.status(Response.Status.SERVICE_UNAVAILABLE).build();
```

Comment:
a 404 should be returned or any other "functionnal" status. a 503 response can be interpreted by a reverse proxy as a backend in difficulty and results to a removal from the pool of functional servers

Reply:
Ok